src/context/auth-context.jsx

```diff
@@ -0,0 +1,26 @@
+import { createContext, useContext, useReducer } from "react";
+import { authReducer } from "../reducer";
+
+const AuthContext = createContext();
+const initialTokenState = localStorage.getItem("token")
+  ? localStorage.getItem("token")
+  : null;
+const initialUserState = localStorage.getItem("token")
+  ? localStorage.getItem("token")
+  : null;
```

Comment:
Why you call local storage two times, you can call one time and put data into intialState ig.